odbc/src/api.rs

```diff
@@ -243,46 +267,64 @@ pub extern "C" fn SQLColumnsW(
 #[no_mangle]
 pub extern "C" fn SQLCompleteAsync(
     _handle_type: HandleType,
-    _handle: Handle,
+    handle: Handle,
     _async_ret_code_ptr: *mut RetCode,
 ) -> SqlReturn {
-    unimplemented!()
+    unsafe {
+        (*(handle as *mut MongoHandle))
+            .add_diag_info(ODBCError::Unimplemented("SQLCompleteAsync".to_string()));
+        SqlReturn::ERROR
+    }
 }
 
 #[no_mangle]
 pub extern "C" fn SQLConnect(
-    _connection_handle: HDbc,
+    connection_handle: HDbc,
     _server_name: *const Char,
     _name_length_1: SmallInt,
     _user_name: *const Char,
     _name_length_2: SmallInt,
     _authentication: *const Char,
     _name_length_3: SmallInt,
 ) -> SqlReturn {
-    unimplemented!()
+    unsafe {
+        (*(connection_handle as *mut MongoHandle))
+            .add_diag_info(ODBCError::Unimplemented("SQLConnect".to_string()));
+        SqlReturn::ERROR
+    }
 }
 
 #[no_mangle]
 pub extern "C" fn SQLConnectW(
-    _connection_handle: HDbc,
+    connection_handle: HDbc,
     _server_name: *const WChar,
     _name_length_1: SmallInt,
     _user_name: *const WChar,
     _name_length_2: SmallInt,
     _authentication: *const WChar,
     _name_length_3: SmallInt,
 ) -> SqlReturn {
-    unimplemented!()
+    unsafe {
+        (*(connection_handle as *mut MongoHandle))
+            .add_diag_info(ODBCError::Unimplemented("SQLConnectW".to_string()));
+        SqlReturn::ERROR
+    }
 }
 
 #[no_mangle]
-pub extern "C" fn SQLCopyDesc(_source_desc_handle: HDesc, _target_desc_handle: HDesc) -> SqlReturn {
-    unimplemented!()
+pub extern "C" fn SQLCopyDesc(source_desc_handle: HDesc, target_desc_handle: HDesc) -> SqlReturn {
+    unsafe {
+        (*(source_desc_handle as *mut MongoHandle))
```

Comment:
I don't think that the error should go in both handles.
From the spec, it should only be added to the TargetDescHandle. See [here](https://docs.microsoft.com/en-us/sql/odbc/reference/syntax/sqlcopydesc-function?view=sql-server-ver15#diagnostics)

> When SQLCopyDesc returns SQL_ERROR or SQL_SUCCESS_WITH_INFO, an associated SQLSTATE value can be obtained by calling SQLGetDiagRec with a HandleType of SQL_HANDLE_DESC and a Handle of TargetDescHandle. If an invalid SourceDescHandle was passed in the call, SQL_INVALID_HANDLE will be returned but no SQLSTATE will be returned.